yark/cli.py

```diff
@@ -205,35 +206,95 @@ def parse_maximum_int(config_arg: str) -> int:
+            )
+            sys.exit(0)
 
-        # Start on archive name
+        # Figure out configuration
+        config = Config()
         if len(args) > 1:
-            # Get name
-            archive_name = args[1]
-
-            # Jank archive check
-            if not Path(archive_name).exists():
-                _err_archive_not_found()
-
-            # Launch and start browser
-            print(f"Starting viewer for {archive_name}..")
-            webbrowser.open(f"http://127.0.0.1:7667/archive/{archive_name}/videos")
-            launch()
-
-        # Start on archive finder
-        else:
-            print("Starting viewer..")
-            webbrowser.open("http://127.0.0.1:7667/")
-            launch()
+
+            def parse_value(config_arg: str) -> str:
+                return config_arg.split("=")[1]
+
+            def parse_port_int(config_arg: str) -> int:
+                """Tries to parse a port number input"""
+                port = parse_value(config_arg)
+                try:
+                    portInt = int(port)
+                    if portInt < 1 or portInt > 65535:
+                        raise Exception("invalid port number")
+                    return portInt
+                except Exception:
+                    print(HELP, file=sys.stderr)
+                    _log_err(
+                        f"\nError: The value '{port}' isn't a valid port number"
+                    )
+                    sys.exit(1)
+
+            # Go through each configuration argument
+            for config_idx in range(1, len(args)):
+                config_arg = args[config_idx]
+
+                # Custom bind host
+                if config_arg.startswith("--host="):
+                    config.bind_host = parse_value(config_arg)
+
+                # Custom bind port
+                elif config_arg.startswith("--port="):
+                    config.bind_port = parse_port_int(config_arg)
+
+                # Disable open in webbrowser behaviour
+                elif config_arg.startswith("--disable-webbrowser"):
+                    config.open_in_webbrowser = False
+
+                # Unknown argument
+                else:
+                    # if config_idx is 1 (first parameter could be name of archive)
+                    if config_idx == 1:
+                        archive_name = config_arg
+                    else:
+                        print(HELP, file=sys.stderr)
+                        _log_err(
+                            f"\nError: Unknown configuration '{config_arg}' provided for archive view"
+                        )
+                        sys.exit(1)
+
+        # Submit config settings
+        config.submit()
+
+        if config.open_in_webbrowser:
```

Comment:
Best to move this to a new function w/ docstring + flow comments

Reply:
I did have an issue reusing `ArchiveNotFoundException` that it would create cyclic references if I used it in the config method.

Could we move the exceptions to a separate directory ?

Reply:
I got rid of the exception, it was adding cruft, I had it there from when I wanted to throw it from the config method and catch it here.

cleaned up and moved the webbrowser opening into config.